c++/src/Compression.cc

```diff
@@ -533,24 +545,37 @@ DIAGNOSTIC_PUSH
   }
 
   /** There are three possible scenarios when seeking a position:
-   * 1. The seeked position is already read and decompressed into
-   *    the output stream.
-   * 2. It is already read from the input stream, but has not been
-   *    decompressed yet, ie. it's not in the output stream.
-   * 3. It is not read yet from the inputstream.
+   * 1. The chunk of the seeked position is already read and decompressed into the output
+   *    stream, ie. chunk header is read and chunk contents are in the output stream.
+   * 2. The chunk of the seeked position is partially read. This only happens for
```

Comment:
The root cause is in line 495 where DecompressionStream::Next only reads availableSize but not the full uncompressed chunk. So another fix is to enforce read full uncompressed chunk there.

Reply:
Yeah, I also thought about that but finally chose this fix for simplicity. Loading the whole uncompressed chunk at once might also improve the performance, we can do it in follow-up tasks. :)